### PR TITLE
syncer(dm): fix startGTID is equal to endGTID

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #4287

### What is changed and how it works?

originally sometimes startGTID is equal to endGTID, but we want to maintain that startLocation <= current event <= endLocation

also remove some log, because I think their effect has been achieved.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: dm/pkg/binlog/position.go [hunk 1]
```diff
@@ -401,9 +401,15 @@ func (l *Location) ResetSuffix() {
...
-	if _, ok := l.gtidSet.(*gtid.MariadbGTIDSet); ok {
+	var flavor string
+
+	switch gset.(type) {
+	case *gmysql.MysqlGTIDSet:
+		flavor = gmysql.MySQLFlavor
+	case *gmysql.MariadbGTIDSet:
 		flavor = gmysql.MariaDBFlavor
+	default:
+		return nil
```
Comment (Ehco1996) [inline dm/pkg/binlog/position.go:412]: this default branch should return err 🤔?

File: dm/syncer/binlog_locations.go [hunk 2]
```diff
@@ -33,7 +34,8 @@ type locationRecorder struct {
 	//       |                       |
 	// curStartLocation        curEndLocation
 	// there may be more events between curStartLocation and curEndLocation due to the limitation of binlog or
-	// implementation of DM, but those events should always belong to one transaction.
+	// implementation of DM, but in such scenario, those events should always belong to one transaction.
+	// When curStartLocation is equal to curEndLocation, it means current event is not a data change.
```
Comment (Ehco1996) [inline dm/syncer/binlog_locations.go:38]: just a question: why `When` is capitalize

Comment (lance6716) [reply dm/syncer/binlog_locations.go:38]: English grammar: capitalize the first letter of a sentence

File: dm/syncer/binlog_locations.go [hunk 3]
```diff
@@ -102,15 +107,39 @@ func shouldUpdatePos(e *replication.BinlogEvent) bool {
...
+	if l.curEndLocation.GetGTID() == nil {
+		gset, _ := gtid.ParserGTID("", gtidStr)
+		_ = l.curEndLocation.SetGTID(gset.Origin())
+		return
+	}
+
+	err = l.curEndLocation.GetGTID().Update(gtidStr)
+	if err != nil {
+		log.L().DPanic("failed to update GTID set",
+			zap.Any("GTID", gtidStr),
```
Comment (GMHDBJD) [inline dm/syncer/binlog_locations.go:138]: zap.String?

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by Ehco1996 — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
/hold

need fix test

**Pull request merged**